networking/external_dns_operator/nw-control-dns-records-public-managed-zone-gcp.adoc

```diff
@@ -0,0 +1,148 @@
++
+[source,terminal]
+----
+$ oc whoami
+----
++
+.Example output
+[source,terminal]
+----
+system:admin
+----
+
+. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json by running the following command:
++
+[source,terminal]
+----
+$ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
+----
+
+. Export Google credentials by running the following command:
++
+[source,terminal]
+----
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+----
+
+. Activate your account by using the following command:
++
+[source,terminal]
+----
+$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+----
+
+. Set your project by running the following command:
++
+[source,terminal]
+----
+$ gcloud config set project <project_id as per decoded-gcloud.json>
+----
+
+. Get the routes to check the domain by running the following command:
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get your zone that installer created for you. Run the following command:
++
+[source,terminal]
+----
+$ gcloud dns managed-zones list | grep qe.gcp.devcluster.openshift.com
+qe-cvs4g-private-zone qe.gcp.devcluster.openshift.com
+----
+
+. Check if the zone only have DNS records either `NS` or `SOA`:
++
+[source,terminal]
+----
+$ gcloud dns record-sets list --zone=qe-cvs4g-private-zone
+----
+
+. Create a `ExternalDNS` CR for `route`. Run the following command:
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp <1>
+spec:
+  domains:
+    - filterType: Include <2>
+      matchType: Exact <3>
+      name: qe.gcp.devcluster.openshift.com <4>
+  provider:
+    type: GCP <5>
+  source:
+    openshiftRouteOptions: <6>
+      routerName: default <7>
+    type: OpenShiftRoute <8>
+----
+
+<1> The `name` defines the name of External DNS CR.
+<2> By default all hosted zones are selected as potential targets. You can include a hosted zone that you need.
+<3> The matching of the target zone's domain has to be exact (as opposed to regular expression match).
+<4> Specify the exact domain of the zone you want to update. The hostname of the routes must be subdomains of the specified domain.
+<5> The `type` field Publishes records on public hosted zone GCP.
```

Comment:
Both public and private hosted zones are supported, so the field `defines Google Cloud DNS provider.`

Reply:
Addressed